src/engine/QueryPlanningCostFactors.cpp

```diff
@@ -24,15 +24,23 @@ QueryPlanningCostFactors::QueryPlanningCostFactors() : _factors() {
   _factors["DISK_RANDOM_ACCESS_COST"] = 1000;
 }
 
+// _____________________________________________________________________________
+
+float toFloat(std::string_view view) {
+  float factor;
+  auto result = std::from_chars(view.data(), view.data() + view.size(), factor);
+  AD_CHECK(!static_cast<int>(result.ec));
```

Comment:
AD_CHECK(...) is for "this should never happen", which obviously is not appropriate for functions that 
get unchecked input from files. Please throw a proper exception with "Could not parse blabla to float when reading QueryPlanningCostFactors from file". And also I think you also have to check, that the complete view was used, because `from_chars("1.23x45")` might not return an error code, but simply `1.23`. (not so sure though, please check and comment)